pkg/graveler/ref/commit_priority_queue.go

```diff
@@ -0,0 +1,58 @@
+package ref
+
+import "github.com/treeverse/lakefs/pkg/graveler"
+
+// CommitsPriorityQueue implements heap.Interface such that the commit with the oldest creation date is at the root
+// of the heap.
+type CommitsPriorityQueue struct {
+	queue []*graveler.CommitRecord
+}
+
```

Comment:
Two suggestions - it will look like less reuse but I think of separation.
1. There are two types here - so split the implementation into two
2. Type the queue without another `struct` level - `type CommitsPriorityQueue []*graveler.CommitRecord`